src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBooster.scala

```diff
@@ -94,7 +130,48 @@ class LightGBMBooster(val model: String) extends Serializable {
         isRowMajor, kind,
         -1, datasetParams, scoredDataLengthLongPtr, scoredDataOutPtr),
       "Booster Predict")
-    predToArray(classification, scoredDataOutPtr, kind)
+    predScoreToArray(classification, scoredDataOutPtr, kind)
+  }
+
+  protected def predictLeafForCSR(sparseVector: SparseVector): Array[Double] = {
+    val numCols = sparseVector.size
+
+    val datasetParams = "max_bin=255 is_pre_partition=True"
```

Comment:
I believe is_pre_partition can be removed here, I don't think it would cause error, I believe it is just unused in this case

Reply:
It is fixed.